ruby-challenges.rb

```diff
@@ -22,10 +37,77 @@
 
 # Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.'
 
+class Bike 
+    attr_accessor :model, :wheels  
+    def initialize(model, wheels,)
+        @model = model
+        @wheels = wheels
+        @mph = 0
+    end
+    
+    # def get_model
+    #     @model
+    # end
+
+    # def get_wheels
+    #     @wheels
+    # end
+
+    def get_mph
+        @mph = 0
+    end 
+
+    def bike_info
+        @bike_info 
+            "the #{@model} has #{@wheels} and is going #{@mph}"
+    end
+
+end
+
 
 
 # -------------------3b) Add the ability to pedal faster and brake. The pedal_faster method should increase the speed by a given amount. The brake method should decrease the speed by a given amount. The bike cannot go negative speeds.
 
+class Bike 
+    attr_accessor :model, :wheels, :current_speed :pedal_faster :brake
+
+    def initialize(model, wheels, current_speed, pedal_faster, brake)
+        @model = model
+        @wheels = wheels
+        @mph = 0
+        @current_speed = current_speed
+        @pedal_faster = pedal_faster
+        @brake = brake
+    end
+
+    def get_mph
+        @mph = 0
+    end 
+
+    def get_current_speed 
+        if @current_speed >= 0
+            p @current_speed
+        elsif 
+            @current_speed = 0
+        end 
+    end
+
+    def pedal_faster
+        @pedal_faster = @current_speed + @pedal_faster
+        p pedal_faster
+    end
+
+    def brake
+        @brake = @current_speed + @brake
+        p brake
+    end
+
+    def get bike_info
+       @bike_info
+        "the #{@model} has #{@wheels} and is going #{@current_speed}"
+    end
+
```

Comment:
One part of this challenge was to prevent the speed from going negative when using the brake method. Also, the pedal_faster and brake methods should expect 1 argument, which would be the number that the speed is either increasing or decreasing by.